### PR TITLE
API to get dashboard data

### PR DESCRIPTION
**Pull request by bhattjai**

### CODE REVIEW

File: datahub/views.py [hunk 1]
```diff
@@ -2390,6 +2389,62 @@ def requested_datasets(self, request, *args, **kwargs):
     #     serializer = self.get_serializer(page, many=True).exclude(is_temp = True)
     #     return self.get_paginated_response(serializer.data)
 
+    @action(detail=True, methods=["get"])
+    def get_dashboard_chart_data(self, request, pk, *args, **kwargs):
+        try:
+            dataset_file = DatasetV2File.objects.get(id=pk)
+            df = pd.read_excel(dataset_file.standardised_file)
```
Comment (i-am-waseem) [inline datahub/views.py:2396]: Handle csv as well.

File: datahub/views.py [hunk 2]
```diff
@@ -2390,6 +2389,62 @@ def requested_datasets(self, request, *args, **kwargs):
...
+                },
+                "insurance_information": {
+                    "insured_crops": (df['Do you insure your crops?']).sum(),
+                    "insured_machinery": (df['Do you insure your farm buildings and other assets?']).sum(),
+                }
+            }
+
+            return Response(
+                obj,
+                status=status.HTTP_500_INTERNAL_SERVER_ERROR,
```
Comment (i-am-waseem) [inline datahub/views.py:2436]: 200 on Success.

File: datahub/views.py [hunk 3]
```diff
@@ -2390,6 +2389,62 @@ def requested_datasets(self, request, *args, **kwargs):
...
+            )
+
+        except DatasetV2File.DoesNotExist:
+            return Response(
+                "No dataset file for the provided id.",
+                status=status.HTTP_500_INTERNAL_SERVER_ERROR,
```
Comment (i-am-waseem) [inline datahub/views.py:2442]: General Exception which is not caught.

File: datahub/views.py [hunk 4]
```diff
@@ -2390,6 +2389,80 @@ def requested_datasets(self, request, *args, **kwargs):
...
+                    "total_no_of_records": len(df),
+                    "male_count": (df['Gender'] == 1).sum(),
+                    "female_count": (df['Gender'] == 0).sum(),
+                    "constituencies": (df['Constituency']).nunique(),
+                    "counties": (df['County']).nunique(),
+                    "sub_counties": (df['Sub County']).nunique(),
+                    "farming_practices": {
+                        "crop_production": (df['Crop Production']).sum(),
+                        "livestock_production": (df['Livestock Production']).sum(),
+                    },
```
Comment (ugeshdg) [inline datahub/views.py:2418]: First get the available columns and do the manipulations on only available columns  , Try to lower all the column names and do the process.

Existing code will work for only positive  flow

File: datahub/views.py [hunk 5]
```diff
@@ -2390,6 +2389,80 @@ def requested_datasets(self, request, *args, **kwargs):
     #     serializer = self.get_serializer(page, many=True).exclude(is_temp = True)
     #     return self.get_paginated_response(serializer.data)
 
+    @action(detail=True, methods=["get"])
+    def get_dashboard_chart_data(self, request, pk, *args, **kwargs):
```
Comment (ugeshdg) [inline datahub/views.py:2393]: Going forward follow TDD.

**Review by ugeshdg — APPROVED** [review]

**Pull request merged**